### ISSUE TITLE
[Linux] Interaction between Moonlight and dbus/Connman introduces network latency by scanning for new wifi APs

### ISSUE THREAD

**Issue by tngit** [moonlight-stream/moonlight-qt#982]
I have choppy sound about every 10 seconds when using Moonlight on wifi, no issues on ethernet.
My first attempt to debug this was running ping on my wifi router while moonlight is running.
This always results in pattern like this (repeating spikes):

64 bytes from 192.168.0.1: icmp_seq=1 ttl=64 time=2.43 ms
64 bytes from 192.168.0.1: icmp_seq=2 ttl=64 time=2.46 ms
64 bytes from 192.168.0.1: icmp_seq=3 ttl=64 time=2.60 ms
64 bytes from 192.168.0.1: icmp_seq=4 ttl=64 time=10.5 ms
64 bytes from 192.168.0.1: icmp_seq=5 ttl=64 time=28.7 ms
64 bytes from 192.168.0.1: icmp_seq=6 ttl=64 time=17.6 ms
64 bytes from 192.168.0.1: icmp_seq=7 ttl=64 time=25.3 ms
64 bytes from 192.168.0.1: icmp_seq=8 ttl=64 time=90.6 ms
64 bytes from 192.168.0.1: icmp_seq=9 ttl=64 time=94.9 ms

With Moonlight not running, the time stays always below 10ms. 
As I first blamed connman, I started connman like this: 
connmand -n -c /etc/connman/main.conf -d

With moonlight running, connman outputs a regular message that it has started a scan (followed by the results): 
plugins/wifi.c:scan_started() 
These messages always appear exactly at the same time I am seeing high latency in ping, and hearing choppy sound.
As soon as I stop Moonlight, those message stop appearing altogether and ping improves. It does not matter if Moonlight is connected to a server or not, it will cause this network performance degradation. 

To really prove this point I restarted connmand, this time with wifi plugin disabled:
connmand -n -c /etc/connman/main.conf -d -P wifi

This causes the following output from Moonlight:

00:00:23 - Qt Warning: propsReply "Method \"GetAll\" with signature \"s\" on interface \"org.freedesktop.DBus.Properties\" doesn't exist\n"
00:00:23 - Qt Warning: nmReply "Method \"GetDevices\" with signature \"\" on interface \"org.freedesktop.NetworkManager\" doesn't exist\n"
00:00:28 - Qt Warning: propsReply "Method \"GetAll\" with signature \"s\" on interface \"org.freedesktop.DBus.Properties\" doesn't exist\n"
00:00:28 - Qt Warning: nmReply "Method \"GetDevices\" with signature \"\" on interface \"org.freedesktop.NetworkManager\" doesn't exist\n"
00:00:31 - Qt Debug: "Method \"Scan\" with signature \"\" on interface \"net.connman.Technology\" doesn't exist\n"
00:00:41 - Qt Debug: "Method \"Scan\" with signature \"\" on interface \"net.connman.Technology\" doesn't exist\n"
00:00:51 - Qt Debug: "Method \"Scan\" with signature \"\" on interface \"net.connman.Technology\" doesn't exist\n"

So whatever the reason, Moonlight seems to hurt it's own performance here. 
I tried disabling the following settings in the Moonlight UI:
- Automatically find PCs on the local network
- Automatically detect blocked connections
This had no effect. 

I did, as a test, stop connman altogether and manually connected to my wifi with just wpa_supplicant. I was absolutely amazed by the performance Sunshine+Moonlight could achieve over wifi and had no issues at all during this test. 

So long story short: I think there is something in Moonlight causing an unnecessary scan of wifi channels about every 10 seconds that harms both general network performance but also its own.

I don't think this helps, but for completeness sake:
The client is running void linux and usiging the moonlight AppImage (though I had the same issue with Flatpak as well, but did not debug it then)
on a notebook with the following components: 
CPU: i5-3427U
GPU: integrated GPU
Wifi: Intel Corporation Centrino Advanced-N 6235

**Comment by cgutman** [comment]
Can you try with the latest nightly? https://ci.appveyor.com/project/cgutman/moonlight-qt/builds/47355116/job/sopbqfwl187uucyc/artifacts

I tried a similar technique to issues reported in the past with QNetworkAccessManager by destroying it when the network operations are completed. https://bugreports.qt.io/browse/QTBUG-40332

**Comment by tngit** [comment]
I tried the nightly build, the results are a bit mixed: Sound is no longer choppy, ping response times remain stable, but the screen stays black after connecting.
I tried connecting using NetworkManager instead of connman and the screen remained black as well. So I assume the screen remaining black is a separate, unrelated issue and you fixed my latency issues. Thank you!

Would you want me to create a separate report about the screen remaining black? Not sure if a bug report against a nightly build is helpful in any way.

**Comment by cgutman** [comment]
If you still have that black screen issue with the latest nightly, please do file a bug.
